### PR TITLE
Fix references to authentication in the docs

### LINKED ISSUE #6855: Fix improper usage of authentication in docs
The documentation about authorization on the Ledger API improperly refers to authentication in many places. Those should be fixed. Unfortunately certain file names and references will need to be kept to not break existing links.

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Fixes #6855

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/app-dev/app-arch.rst [hunk 1]
```diff
@@ -92,18 +92,18 @@ If you choose a different JavaScript based frontend framework, the packages ``@d
...
 your users.
 
-DAML ledgers support a unified interface for authentication of commands. Some DAML ledgers like for
-example https://projectdabl.com offer an integrated authentication service, but you can also use an
-external service provider for authentication like https://auth0.com. The DAML react libraries
-support interfacing with an authenticated DAML ledger. Simply initialize your ``DamlLedger`` object
-with the token obtained by an authentication service. How authentication works and the form of the
-required tokens is described in the :ref:`Authentication <authentication>` section.
+DAML ledgers support a unified interface for authorization of commands. Some DAML ledgers like for
+example https://projectdabl.com offer integrated authentication and authorization, but you can also
```
Comment (gerolf-da) [inline docs/source/app-dev/app-arch.rst:102]:
```suggestion
DAML ledgers support a unified interface for authorization of commands. Some DAML ledgers, like for
example https://projectdabl.com, offer integrated authentication and authorization, but you can also
```

File: docs/source/app-dev/app-arch.rst [hunk 2]
```diff
@@ -92,18 +92,18 @@ If you choose a different JavaScript based frontend framework, the packages ``@d
...
+use an external service provider like https://auth0.com. The DAML react libraries support interfacing
+with a DAML ledger that validates authorization of incoming requests. Simply initialize your
+``DamlLedger`` object with the token obtained by the proper issuer. How authorization works and the
```
Comment (gerolf-da) [inline docs/source/app-dev/app-arch.rst:105]:
```suggestion
``DamlLedger`` object with the token obtained by the respective token issuer. How authorization works and the
```

File: docs/source/app-dev/authorization.rst [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 
-Authentication
-##############
+Authorization
+#############
 
 When developing DAML applications using SDK tools,
-your local setup will most likely not use any authentication -
+your local setup will most likely not verify any authorization -
 by default, any valid ledger API request will be accepted by the sandbox.
```
Comment (gerolf-da) [inline docs/source/app-dev/authorization.rst:11]:
```suggestion
by default. Any valid ledger API request will be accepted by the sandbox.
```

File: docs/source/app-dev/authorization.rst [hunk 4]
```diff
@@ -61,8 +61,11 @@ Glossary:
 - A ``token`` (or ``access token``) is a tamper-proof piece of data that contains security information, such as the user identity or its privileges.
 - A ``token issuer`` is a service that generates tokens. Also known as "authentication server" or "Identity and Access Management (IAM) system".
 
+.. this anchor is kept around to not break existing links
```
Comment (gerolf-da) [inline docs/source/app-dev/authorization.rst:64]: Is the anchor also rendered in HTML?

File: docs/source/app-dev/authorization.rst (hunk 3 above)
Comment (stefanobaghino-da) [reply docs/source/app-dev/authorization.rst:11]: Note the `-` before. This is a single sentence that states the default behavior, which is to accept any valid request without verifying its authorization.

File: docs/source/app-dev/authorization.rst (hunk 4 above)
Comment (stefanobaghino-da) [reply docs/source/app-dev/authorization.rst:64]: It doesn't, thanks for asking. I'll revert all such changes.

Comment (stefanobaghino-da) [reply docs/source/app-dev/authorization.rst]: Addressed by https://github.com/digital-asset/daml/pull/6858/commits/a40534f8feb3aa8d303159b5f9823204f7687a35

**Pull request merged**